pkg/pool-manager/pool.go

```diff
@@ -147,6 +150,15 @@ func (p *PoolManager) InitMaps() error {
 	return nil
 }
 
+func (p *PoolManager) IsMacInRange(macAddress string) bool {
+	macAddressHW, _ := net.ParseMAC(macAddress)
+	return inRange(p.rangeStart, p.rangeEnd, macAddressHW)
+}
+
+func (p *PoolManager) GetManagerFinalizerName() string {
+	return fmt.Sprintf("%s%s", RuntimeObjectFinalizerName, p.managerPodName)
```

Comment:
Using pod name as the finalizer name is dangerous. It basically forces us never to rename the manager pod, otherwise there would be forever trailing finalizer. Can we use the index of the range instead? Or maybe better (index-numOfReplicas), so we can calculate what part of range was allocated.

Reply:
afaik, the index is best attained inside a pod - from it's name. Good catch, thanks!